package/utils/executeStringFunction.ts

```diff
@@ -53,6 +68,32 @@ export const executeStringFunction = async (
       // Call the function with the parsed arguments
       // @ts-expect-error later
       const result = await module[functionName](...args);
+      if (isSmartContractResponse(result)) {
```

Comment:
Let's take this part out in same file if possible. it will be clear to read it in future may be.